packages/samples/headless-react/src/components/query-summary/query-summary.fn.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildQuerySummary,
+  QuerySummary as HeadlessQuerySummary,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface QuerySummaryProps {
+  controller: HeadlessQuerySummary;
+}
+
+export const QuerySummary: FunctionComponent<QuerySummaryProps> = (props) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  return state.hasQuery ? (
+    <p>
+      Results {state.firstResult}-{state.lastResult} of{' '}
+      <strong>{state.total}</strong> for <strong>{state.query}</strong> in{' '}
+      {state.durationInSeconds.toLocaleString()} seconds
```

Comment:
Same comment as above to simplifying. Otherwise, I like how short this sample is. I would say 24-30 lines is a good target to aim for with functional components which are more compact.